### PR TITLE
Cut various dependencies out of the oc binary

### PR DESCRIPTION
**Pull request by smarterclayton**
We were pulling in kubelet, more of the oauth server. This cuts total binary size by about 5M

### CODE REVIEW

**Comment by mfojtik** [comment]
LGTM (need to fix unit test)

File: pkg/oc/admin/diagnostics/diagnostics/cluster/network/in_pod/util/log.go [hunk 1]
```diff
@@ -206,3 +206,40 @@ func (l *LogInterface) getConfigFileForService(serviceName, serviceArgs string)
 	}
 	return string(out[:])
 }
+
+// ContainerID is a type that identifies a container.
+type ContainerID struct {
```
Comment (soltysh) [inline pkg/oc/admin/diagnostics/diagnostics/cluster/network/in_pod/util/log.go]: Any particular reason we're copy&pasting this from upstream? The functionality is *exactly* the same.

**Comment by smarterclayton** [comment]
In general, anyone who is taking a dependency across component lines (kubelet, CLI, apiserver, controller manager) has to have a justification.  Code duplication is not a sufficient justification in most cases.  So the burden of proof on the implementer is 

1. is the code that is shared a common API?  in which case the code should either be moved to an API package, or can be safely duplicated, because there is an API boundary in place.
2. if the code is not a common API, then we are relying on internal details that could break when components differ, and the code needs to be changed to accommodate that, or a test needs to be in place to verify that they are not broken, and the presence of the test is the important part, not sharing code.

The dependencies I'm snipping have never been acceptable, and their presence was always temporary.

File: pkg/oc/admin/diagnostics/diagnostics/cluster/network/in_pod/util/log.go (hunk 1 above)
Comment (soltysh) [reply pkg/oc/admin/diagnostics/diagnostics/cluster/network/in_pod/util/log.go]: > quasi public packages

That's sufficiently strong option.

**Comment by smarterclayton** [comment]
Fixed a bad import that crept in, retagging.

**Pull request merged**